examples/autoreply/src/lib.rs

```diff
@@ -58,7 +61,9 @@ mod tests {
         assert!(!res.main_failed());
 
         // Check whether the auto-reply was received
-        let reply_received: bool = prog2.read_state().expect("Failed to read state");
+        let reply_received: bool = prog2
+            .read_state(Vec::<u8>::default())
```

Comment:
I'd use just `vec![]`
```suggestion
            .read_state(vec![])
```

Reply:
If I do that it cannot infer the type, since the function accepts a E: Encode, it doesn't know what type the Vec should contain, that's why I use Vec::<u8>::default()

Reply:
That's the reason to consider passing Option in Encodable fn (to simplify default arg). How do you feel about it?

Reply:
Changed to use Option, if `None` is passed, an empty `Vec<u8>` is used.

Reply:
Reverted due to an issue with generics, if you just pass `None`, it will not compile and say that it cannot infer the type of P.